app/vmalert/remotewrite/remotewrite.go

```diff
@@ -276,22 +275,21 @@ func (c *Client) send(ctx context.Context, data []byte) error {
 	case 2:
 		// respond with a HTTP 2xx status code when the write is successful.
 		return nil
-	case 5:
-		// respond with HTTP status code 5xx when the write fails and SHOULD be retried.
-		return &retriableError{fmt.Errorf("unexpected response code %d for %s. Response body %q",
-			resp.StatusCode, req.URL.Redacted(), body)}
-	default:
+	case 4:
 		// respond with HTTP status code 4xx when the request is invalid, will never be able to succeed
 		// and should not be retried.
+		return &nonRetriableError{fmt.Errorf("unexpected response code %d for %s. Response body %q",
```

Comment:
There is `http code 429 Too Many Requests` which code be retriable. Should we handle this case?

Reply:
We process errors here according to https://prometheus.io/docs/concepts/remote_write_spec/#retries-backoff 👍 
```
 They MUST respond with HTTP status code 4xx when the request is invalid, will never be able to succeed and should not be retried.
```

Reply:
Yes, but there is also a note:
```
They MAY retry on HTTP 429 responses, which could result in senders "falling behind" if the server cannot keep up
```
 And vmauth could return a 429 http code.

Reply:
Yes, I missed that. Will update